### PR TITLE
Work/animation prod

### PR DESCRIPTION
**Pull request by lilianabulgaru3pg**

### CODE REVIEW

File: SweetsFactory/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "core-js": "^2.5.0",
     "firebase": "^4.2.0",
     "foundation-sites": "^6.4.4-rc1",
-    "jquery": "^3.2.1",
+    "jquery": "^3.2.0",
```
Comment (fadrian9006) [inline SweetsFactory/package.json:23]: Which was the reason for changing the jQuery version?

File: SweetsFactory/src/app/shared/product/productItem.component.ts [hunk 2]
```diff
@@ -1,13 +1,49 @@
...
+                animate(250)
+            ])
+        ])
+    ]
 })
 
 export class ProductItemComponent {
     @Input() product: Product;
-    constructor() { }
+    state = 'inactive';
```
Comment (fadrian9006) [inline SweetsFactory/src/app/shared/product/productItem.component.ts:38]: What is the purpose of  the "state" variable in this case?

File: SweetsFactory/src/app/shared/product/productItem.component.html [hunk 3]
```diff
@@ -1,5 +1,9 @@
-<div class="item">
-    <img class="img-15 float-center" src="../images/icecream.png" alt="">
-    <p class="text-center text-item">{{product.name}}</p>
-    <p class="text-center text-item">{{product.price}} ron</p>
+<div class="item" (click)="toggleState()">
+    <img class="img-15 float-center" [src]="imagePath" alt="">
+    <div [@prodItemState]="state" class="state-div" [ngStyle]="{'position': state ==='active' ?
```
Comment (fadrian9006) [inline SweetsFactory/src/app/shared/product/productItem.component.html:3]: What "state-div" means/does?

File: SweetsFactory/src/app/shared/product/productItem.component.html [hunk 4]
```diff
@@ -1,5 +1,9 @@
...
+    'absolute' : 'relative'}">
+        <p class="text-center text-item">{{product.name}}</p>
+        <p class="text-center text-item">{{product.price}} ron</p>
+        <button [ngClass]="{'hide': state ==='inactive', 'hollow button shop-btn': true}">SHOP</button>
```
Comment (fadrian9006) [inline SweetsFactory/src/app/shared/product/productItem.component.html:7]: Why do you need this << 'hollow button shop-btn': true >>  inside ngClass directive?

File: SweetsFactory/package.json (hunk 1 above)
Comment (lilianabulgaru3pg) [reply SweetsFactory/package.json:23]: I had some issues with using $, i don't think that was the fix to that but i left it that way, but finally worked after an npm instal

File: SweetsFactory/src/app/shared/product/productItem.component.ts (hunk 2 above)
Comment (lilianabulgaru3pg) [reply SweetsFactory/src/app/shared/product/productItem.component.ts:38]: I switch trough states when animating the overlay for a product

File: SweetsFactory/src/app/shared/product/productItem.component.html (hunk 3 above)
Comment (lilianabulgaru3pg) [reply SweetsFactory/src/app/shared/product/productItem.component.html:3]: It's the overlay

File: SweetsFactory/src/app/shared/product/productItem.component.html (hunk 4 above)
Comment (lilianabulgaru3pg) [reply SweetsFactory/src/app/shared/product/productItem.component.html:7]: ng-class and class can be used together or just ng-class

File: SweetsFactory/src/app/shared/product/productItem.component.html (hunk 3 above)
Comment (fadrian9006) [reply SweetsFactory/src/app/shared/product/productItem.component.html:3]: I recommend to use more appropriate names

File: SweetsFactory/src/app/shared/product/productItem.component.html (hunk 4 above)
Comment (fadrian9006) [reply SweetsFactory/src/app/shared/product/productItem.component.html:7]: I see, of course they should work.
Do you need to pass those classes through directive?

File: SweetsFactory/src/app/shared/product/productItem.component.ts (hunk 2 above)
Comment (fadrian9006) [reply SweetsFactory/src/app/shared/product/productItem.component.ts:38]: You don't need javascript logic for doing such animations.

**Pull request merged**